examples/on_gke/new_gke_cluster/main.tf

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//*****************************************
+//  Setup Google providers
+//*****************************************
+
+provider "google" {
+  version = "~> 2.7.0"
+  credentials = "${file(var.credentials_path)}"
+}
+provider "google-beta" {
+  version = "~> 2.7.0"
+  credentials = "${file(var.credentials_path)}"
+}
+
+//*****************************************
+//  Setup the Kubernetes Provider
+//*****************************************
+
+data "google_client_config" "default" {}
+
+provider "kubernetes" {
+  load_config_file = false
+
+  host = "https://${module.gke.endpoint}"
```

Comment:
Has this configuration been tested? It seems like it could cause a recursive issue. Since there is only one Kubernetes provider configured, it would be used for the `gke` module, and the `endpoint` output comes from the `google_container_cluster` resource which I assume would require a configured provider. My thought is we will need two Kubernetes providers, this one to be aliased and explicitly passed to the `forseti-on-gke` module, and an other to be explicitly passed to the `gke` module.

Reply:
Yes, I have tested it. The [terraform-google-modules/terraform-google-kubernetes-engine](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/d8e78504c51b194ce30e736f7a3a4240b9dbcd9d/auth.tf#L26-L34) module creates it's own Kubernetes provider.  I don't think one has to be passed in their.  In any case I think I understand what you are saying.

Reply:
@aaron-lane I think I've got this.  PTAL/